Added argument summary prompt text

david-vectara opened the pull request:
As requested by Tallat, I've added the prompt text argument which allows customers to test out different prompts, potentially including metadata fields.

Please be aware that due to typescript property parsing, the $ in the custom prompt must be escaped with \$.

\" will be converted to \\\" and \n (string not newline) will be converted to real newline.

ofermend commented on src/contexts/sendSearchRequest.ts:
```diff
@@ -78,6 +86,7 @@ export const sendSearchRequest = async ({
                   responseLang: language,
                   maxSummarizedResults: summaryNumResults,
                   summarizerPromptName: summaryPromptName,
+                  promptText: summaryPromptText
```
Does this work?
In other words, if summaryPromptText is just an empty string does it get ignored in the backend or would it fail? Perhaps instead check if it's empty and if so then just not add it here?

david-vectara commented:
Yes, works both with and without the property.

I've also added a defensive check before doing a "replaceAll" to convert
the string literal "\n" into a newline.

Here's a test prompt if you want to try it (built for TradieCart) use case
we're working on.

REACT_APP_summary_prompt_text=[ \n {"role": "system", "content": "You are a
search bot that takes search results and provides an answer to query:
\"\$esc.java(\${vectaraQuery})\". Only use information provided in this
chat. Provide any relevant qualifying instructions that may also be
relevant to the answer."}, \n {"role": "user", "content": "From the
following information, answer the question,
\"\$esc.java(\${vectaraQuery}).\"\n\n #foreach (\$qResult in
\$vectaraQueryResults) \n \$foreach.index
\n\$qResult.docMetadata().get("manufacturer")
\$qResult.docMetadata().get("product")\n\n
\$esc.java(\${qResult.getText()}) \n  #end "} ]

On Sat, Mar 2, 2024 at 2:16 PM Ofer Mendelevitch ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/contexts/sendSearchRequest.ts
> <https://github.com/vectara/vectara-answer/pull/138#discussion_r1509874920>
> :
>
> > @@ -78,6 +86,7 @@ export const sendSearchRequest = async ({
>                    responseLang: language,
>                    maxSummarizedResults: summaryNumResults,
>                    summarizerPromptName: summaryPromptName,
> +                  promptText: summaryPromptText
>
> Does this work?
> In other words, if summaryPromptText is just an empty string does it get
> ignored in the backend or would it fail? Perhaps instead check if it's
> empty and if so then just not add it here?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vectara/vectara-answer/pull/138#pullrequestreview-1912566236>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BETWLRVZQQO35IPQ56CX5T3YWFABRAVCNFSM6AAAAABECVDXA2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMJSGU3DMMRTGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
David Levy
Head of Field Engineering, APAC
m: +61 438 442 286, e: ***@***.***
Vectara - *Improving the Efficiency of Knowledge Workers*

david-vectara commented:
Yes, works both with and without the property.

I've also added a defensive check before doing a "replaceAll" to convert
the string literal "\n" into a newline.

On Sat, Mar 2, 2024 at 3:37 PM David Levy ***@***.***> wrote:

> Yes, works both with and without the property.
>
> I've also added a defensive check before doing a "replaceAll" to convert
> the string literal "\n" into a newline.
>
> Here's a test prompt if you want to try it (built for TradieCart) use case
> we're working on.
>
> REACT_APP_summary_prompt_text=[ \n {"role": "system", "content": "You are
> a search bot that takes search results and provides an answer to query:
> \"\$esc.java(\${vectaraQuery})\". Only use information provided in this
> chat. Provide any relevant qualifying instructions that may also be
> relevant to the answer."}, \n {"role": "user", "content": "From the
> following information, answer the question,
> \"\$esc.java(\${vectaraQuery}).\"\n\n #foreach (\$qResult in
> \$vectaraQueryResults) \n \$foreach.index
> \n\$qResult.docMetadata().get("manufacturer")
> \$qResult.docMetadata().get("product")\n\n
> \$esc.java(\${qResult.getText()}) \n  #end "} ]
>
> On Sat, Mar 2, 2024 at 2:16 PM Ofer Mendelevitch ***@***.***>
> wrote:
>
>> ***@***.**** commented on this pull request.
>> ------------------------------
>>
>> In src/contexts/sendSearchRequest.ts
>> <https://github.com/vectara/vectara-answer/pull/138#discussion_r1509874920>
>> :
>>
>> > @@ -78,6 +86,7 @@ export const sendSearchRequest = async ({
>>                    responseLang: language,
>>                    maxSummarizedResults: summaryNumResults,
>>                    summarizerPromptName: summaryPromptName,
>> +                  promptText: summaryPromptText
>>
>> Does this work?
>> In other words, if summaryPromptText is just an empty string does it get
>> ignored in the backend or would it fail? Perhaps instead check if it's
>> empty and if so then just not add it here?
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/vectara/vectara-answer/pull/138#pullrequestreview-1912566236>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/BETWLRVZQQO35IPQ56CX5T3YWFABRAVCNFSM6AAAAABECVDXA2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMJSGU3DMMRTGY>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>
>
> --
> David Levy
> Head of Field Engineering, APAC
> m: +61 438 442 286, e: ***@***.***
> Vectara - *Improving the Efficiency of Knowledge Workers*
>


-- 
David Levy
Head of Field Engineering, APAC
m: +61 438 442 286, e: ***@***.***
Vectara - *Improving the Efficiency of Knowledge Workers*

ofermend reviewed (approved):
LGTM
